Documentation/cli-installation-scenarios.md

```diff
@@ -0,0 +1,237 @@
+| CentOS/RH           	| RPM              	| Though it currently doesn't exist, we should look into enabling RPMs for CentOS/RH-based distributions. This would bring it to par with Debian ones.                                                                                               	|
+
+
+## Installation script via code snippet
+This approach is a shell one-liner that downloads an installation script and runs it. The installation script will download the latest zip/tarball (depending on the script used) and will unpack it to a given location. After that, the script will print out what needs to be set for the entire CLI to work (env variables, $PATH modification etc.).
+
+This install covers the following main scenario: 
+
+* Local installation on a dev machine
+* Acquiring tools on a CI build
+
+  
+The features the script needs to support/have are:
+* Support for dev and stable channel
+* Support for specifying the version
+* Support for specfying the installation location
+
+
+### Local installation
+The local installation puts the bits in any folder on disk that the user specifies when invoking the script. The layout is pretty much the same as with native installers.
+
+The local install can "grow up" into a global one simply by pointing the system PATH to the relevant bin folder within the layout on disk. 
+
+Acquiring the tools on the CI server would be done in the same way as on a user machine with a local install: the install script would be invoked with a given set of options and then further build scripts would actually use the tools installed in this way to do their thing (build, restore, pack/publish etc.)
+
+The guidance is, of course, to always use the beta channel for the script, and this is what the script will have as the default option. 
+
+### Installation script features
+The following arguments are needed for the installation script:
+
+>--channel / -Channel
+>Which channel (i.e. "dev", "beta", "release") to install from; defaults to beta if not present. 
+
+>--version / -Version
+>Which version of CLI to install; you need to specify the version as 3-part version (i.e. 1.0.0-13232323); defaults to latest if not present. 
+
+>--prefix / -InstallDir
+>Path to where to install the CLI bundle; defaults to /.dotnet if not present.
+
+**TODO: how do you call the script**
+
+### Windows command example 
+
+```
+@powershell -NoProfile -ExecutionPolicy unrestricted -Command "&{iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/install.ps1'))}"
+```
+
+### OSX/Linux shell command
+
+```
+curl -sSL https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/install.sh | bash /dev/stdin [args] 
+```
+
+## Complete manual installation
```

Comment:
Are we sure we need this? I'd rather not document the specific URLs that contain every build... when would someone not use a script?